pkg/spi-shared/metrics/server.go

```diff
@@ -0,0 +1,61 @@
+package metrics
+
+import (
+	"context"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"os"
+	"os/signal"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+//var metricsRegistry *prometheus.Registry
+
+func ServeMetrics(ctx context.Context, address string) {
+	setupLog := ctrl.Log.WithName("metrics")
+	//metricsRegistry = prometheus.NewRegistry()
+
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", promhttp.InstrumentMetricHandler(
+		prometheus.DefaultRegisterer, promhttp.HandlerFor(prometheus.DefaultGatherer, promhttp.HandlerOpts{})))
+	server := &http.Server{Addr: address,
```

Comment:
Why do we need to start 2 different HTTP servers? - one for our endpoints and the other for metrics?

Reply:
To answer myself - we need 2 servers, because metric collection happens on a different port due to the need to separate the auth with kube-rbac-proxy.